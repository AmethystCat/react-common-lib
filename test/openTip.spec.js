import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Icon } from 'antd';
import OpenTip from '../components/openTip';

Enzyme.configure({ adapter: new Adapter() });

describe('OpenTip UT', () => {
  test('snapshot', () => {
    const component = renderer.create(<OpenTip content="test" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should_show_modal_which_include_given_content_when_click_icon', () => {
    const txt = '哈哈';
    const content = <div id="content">{txt}</div>;
    const wrapper = mount(<OpenTip {...{ content }} />);
    const icon = wrapper.find(Icon);

    icon.simulate('click');
    const hasRenderModalNode = window.document.querySelector('.ant-modal');
    const contentNode = window.document.querySelector('#content');
    
    expect(hasRenderModalNode).toBeTruthy();
    expect(contentNode).toBeTruthy();
    expect(contentNode.innerHTML).toEqual(txt);
  });
});
