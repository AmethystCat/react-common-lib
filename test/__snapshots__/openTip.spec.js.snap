// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenTip UT snapshot 1`] = `
<div
  className="section-open-tip"
>
  <i
    className="anticon anticon-info-circle-o"
    onClick={[Function]}
  />
</div>
`;

exports[`OpenTip UT snapshot 2`] = `
<div
  className="section-open-tip"
>
  <i
    className="anticon anticon-info-circle-o"
    onClick={[Function]}
  />
</div>
`;
